lib/ffmpeg/raw_command_args.rb

```diff
@@ -0,0 +1,775 @@
+        arg("bsf:#{type.to_s[0]}", Filter.join(*group))
+      end
+
+      self
+    end
+
+    # Adds a new filter complex to the command arguments.
+    #
+    # @param filters [Array<FFMPEG::Filter, String>] The filters to add.
+    # @return [self]
+    def filter_complex(*filters)
+      arg('filter_complex', filters.compact.map(&:to_s).join(';'))
+
+      self
+    end
+
+    # Sets the output format in the command arguments.
+    #
+    # @param value [String] The format to set.
+    # @return [self]
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #   output_format 'dash'
+    #  end
+    #  args.to_s # "-f dash"
+    def format_name(value)
+      arg('f', value)
+    end
+
+    # Adds new muxing flags in the command arguments.
+    #
+    # @param value [String] The flags to add.
+    # @return [self]
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #    muxing_flags '+faststart+frag_keyframe'
+    #  end
+    #  args.to_s # "-movflags +faststart+frag_keyframe"
+    def muxing_flags(value)
+      arg('movflags', value)
+    end
+
+    # Sets the buffer size in the command arguments.
+    #
+    # @param value [String, Numeric] The buffer size to set.
+    # @return [self]
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #    buffer_size '64k'
+    #  end
+    #  args.to_s # "-bufsize 64k"
+    def buffer_size(value)
+      arg('bufsize', value)
+    end
+
+    # Sets the duration in the command arguments.
+    #
+    # @param value [String, Numeric] The duration to set.
+    # @return [self]
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #    duration 10
+    #  end
+    #  args.to_s # "-t 10"
+    def duration(value)
+      arg('t', value)
+    end
+
+    # Sets the segment duration in the command arguments.
+    # This is used for adaptive streaming.
+    #
+    # @param value [String, Numeric] The segment duration to set.
+    # @return [self]
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #    segment_duration 2
+    #  end
+    #  args.to_s # "-seg_duration 2"
+    def segment_duration(value)
+      arg('seg_duration', value)
+    end
+
+    # Sets a constant rate factor in the command arguments.
+    #
+    # @param value [String, Numeric] The constant rate factor to set.
+    # @param kwargs [Hash] The stream specific arguments to use (see stream_arg).
+    # @return [self]
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #    constant_rate_factor 23
+    #  end
+    #  args.to_s # "-crf 23"
+    #
+    # @example
+    #  args = FFMPEG::RawCommandArgs.compose do
+    #    constant_rate_factor 23, stream_id: 'v:0'
+    #  end
+    #  args.to_s # "-crf:v:0 23"
+    def constant_rate_factor(value, **kwargs)
+      stream_arg('crf', value, **kwargs)
+    end
+
+    # =================== #
+    # === VIDEO UTILS === #
```

Comment:
nit: should we move these into their own helper modules?

```ruby
module FFMPEg
  class RawCommandArgs
    module VideoUtils
      def video_codec_name(...)
         ...
      end
     
       # and all other video-only utils
    end
  end
end
```

and then we can just `include RawCommandArgs::VideoUtils` here to have smaller files